objects/src/notes/script.rs

```diff
@@ -29,3 +29,37 @@ impl NoteScript {
         &self.code
     }
 }
+
+/// NoteScriptInputs holds the inputs that are placed on the stack at the initiation of note script
+/// execution.
+#[derive(Debug, Eq, PartialEq)]
+pub struct NoteScriptInputs {
+    inputs: Vec<Felt>,
+    hash: Digest,
+}
+
+impl NoteScriptInputs {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates a NoteScriptInputs struct from an array of Felt's
+    pub fn new(inputs: &[Felt]) -> Self {
+        Self {
+            inputs: inputs.to_vec(),
+            hash: Hasher::hash_elements(inputs),
+        }
+    }
```

Comment:
I think we should limit the number of inputs to 16. The main reasons is that when `call` instruction executed, only the top 16 elements of the stack area available to the callee.

We may even say that note inputs always consist of 16 elements to simplify note setup script.